### PR TITLE
Add a gist of the difference between code-server and Coder

### PR DESCRIPTION
**Pull request by im-coder-lg**
Fixes #

Here @bpmct I did what I could. Looping in other people who need to review this:

@jsjoeio
@tmikaeld

And all maintainers.

Thanks to @tmikaeld for a small idea of the difference!

### CODE REVIEW

**Review by jsjoeio — APPROVED** [review]
This looks good to me!

File: docs/README.md [hunk 1]
```diff
@@ -13,6 +13,9 @@ access it in the browser.
 - Use cloud servers to speed up tests, compilations, downloads, and more
 - Preserve battery life when you're on the go; all intensive tasks run on your
   server
+  
+| 🔔 code-server is the free and libre single-user variant of [Coder](https://coder.com/), our enterprise developer workspace platform. For more information, visit [Coder.com](https://coder.com/docs/comparison)
```
Comment (jsjoeio) [inline docs/README.md:17]: What is the `|` for? Do we need that?

**Comment by jsjoeio** [comment]
I'll let @bpmct give the final 👍

File: docs/README.md (hunk 1 above)
Comment (im-coder-lg) [reply docs/README.md:17]: Yeah, idk but I think it helps making the box.

Comment (im-coder-lg) [reply docs/README.md:17]: Wait, I understood! In Markdown, you need the `|` to make a table, right? We made a 1x1 table aka. the banner that has the alert! In Markdown, you make tables like this:
```markdown
|      |      |      |
| ---- | ---- | ---- |
|      |      |      |
|      |      |      |
|      |      |      |
```

How it looks:
|      |      |      |
| ---- | ---- | ---- |
|      |      |      |
|      |      |      |
|      |      |      |


So, we used this:
```markdown
| Hello
| ---
```
And it looks like this:
| Hello
| ---

Do you get it?

**Comment by im-coder-lg** [comment]
Some checks are failing since some things like tokens are failing, what should fix that?

**Comment by im-coder-lg** [comment]
Wait, why's this on the 3.12.2 milestone? This has to be on the 3.12.1 milestone, right? Or, will this be merged after 3.12.1?

File: docs/README.md (hunk 1 above)
Comment (jsjoeio) [reply docs/README.md:17]: OH! That makes sense. Let's keep it then 👍  Thanks for the explanation and demo :)

**Comment by jsjoeio** [comment]
> Some checks are failing since some things like tokens are failing, what should fix that?

The Docs preview is broken for forks unfortunately so you can ignore that (we will fix hopefully soon). The Build/Pre-build shouldn't be failing though 🤔 I just re-ran it

**Comment by jsjoeio** [comment]
This makes no sense  🤔 @code-asher or @TeffenEllis any ideas why the Build is failing [here](https://github.com/cdr/code-server/runs/4084268809?check_suite_focus=true#step:5:331)

```logs
Traceback (most recent call last):
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 50, in <module>
    sys.exit(gyp.script_main())
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 554, in script_main
    return main(sys.argv[1:])
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 547, in main
    return gyp_main(args)
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 520, in gyp_main
    [generator, flat_list, targets, data] = Load(
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 136, in Load
    result = gyp.input.Load(build_files, default_variables, includes[:],
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 2782, in Load
    LoadTargetBuildFile(build_file, data, aux_data,
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 391, in LoadTargetBuildFile
    build_file_data = LoadOneBuildFile(build_file_path, data, aux_data,
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 263, in LoadOneBuildFile
    LoadBuildFileIncludesIntoDict(build_file_data, build_file_path, data,
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 301, in LoadBuildFileIncludesIntoDict
    LoadOneBuildFile(include, data, aux_data, None, False, check),
  File "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 243, in LoadOneBuildFile
    build_file_data = eval(build_file_contents, {'__builtins__': {}},
  File "/home/runner/.cache/node-gyp/14.16.0/include/node/common.gypi", line 43
    # Turn on SipHash for h
                          ^
SyntaxError: unexpected EOF while parsing
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:282:12)
gyp ERR! System Linux 5.8.0-1042-azure
gyp ERR! command "/opt/hostedtoolcache/node/14.18.1/x64/bin/node" "/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/runner/work/code-server/code-server/vendor/modules/code-oss-dev/remote/node_modules/node-pty
gyp ERR! node -v v14.18.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
```

**Comment by jsjoeio** [comment]
Okay this is easy. Can you run `yarn fmt`, commit changes, rebase and then push up? That should fix this:

image

File: docs/README.md (hunk 1 above)
Comment (jsjoeio) [inline docs/README.md:17]: On second thought, can you change this to this?

```suggestion
| 🔔 code-server is a free browser-based IDE while [Coder](https://coder.com/), is our enterprise developer workspace platform. For more information, visit [Coder.com](https://coder.com/docs/comparison)
```

**Review by jsjoeio — CHANGES_REQUESTED** [review]
Minor change + `yarn fmt`

**Comment by im-coder-lg** [comment]
Will do that soon, need to get my laptop out.

**Comment by im-coder-lg** [comment]
Sorry for the delay @jsjoeio it will be here soon, I just had a bunch of problems with Gitpod, the main one being my email, so it took some time to straighten things up, but I did `yarn fmt` before, and I will need to do it again.

**Comment by im-coder-lg** [comment]
Here this is... Is this correct?

**Comment by im-coder-lg** [comment]
@jsjoeio just curious... The README is in the docs, then how did you loop it into the repo homepage? I mean, that looks impossible, right?

**Comment by jsjoeio** [comment]
> just curious... The README is in the docs, then how did you loop it into the repo homepage? I mean, that looks impossible, right?

GitHub let's you put it at the root or in `/docs`. See here:

> If you put your README file in your repository's root, docs, or hidden .github directory, GitHub will recognize and automatically surface your README to repository visitors.

[Link](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes)

**Comment by jsjoeio** [comment]
@im-coder-lg sorry, do you mind rebasing one more time? That should fix the e2e check and then I can merge!

**Comment by im-coder-lg** [comment]
Wait, you mean running `yarn fmt` again?

**Comment by im-coder-lg** [comment]
https://user-images.githubusercontent.com/76845820/140596967-5870731b-4990-42ea-8795-1ff86c402b56.mp4

The video up there is the problem I have with `yarn fmt`. It does work, but then disappears(changes). Do you know what is happening with this?

And yeah, I am using Gitpod because my WIndows PC is slow. It is Windows 7, meaning I use Node 12. So I use Gitpod, and I switched from Node 16 to Node 14 via nvm as shown above. I have the latest yarn version but to no avail. Anybody, any ideas?

**Comment by jsjoeio** [comment]
> Wait, you mean running yarn fmt again?

No, rebasing so you have the latest changes from `main`. This is usually how I do it:

1. `git rebase origin/main`
2. `git push --force-with-lease`

But since you're on a fork, you'll probably ned to modify to `git rebase upstream/main`

**Comment by im-coder-lg** [comment]
I need to fix my brain. I am a Git pro but didn't know the simple thing called rebasing 😂 but thanks for explaining, I will try that now. Should I run `yarn fly again? Maybe I should so doing that now!

**Comment by jsjoeio** [comment]
> I need to fix my brain. I am a Git pro but didn't know the simple thing called rebasing 😂 but thanks for explaining, I will try that now.

lol no worries!

> Should I run `yarn fly again? Maybe I should so doing that now!

Nope, no need! Rebasing just get you in line with `main` and the rest of CI should pass (minus Docs preview)

**Comment by im-coder-lg** [comment]
Heads up! Rebase failed. The console logs are here:

[asciicast](https://asciinema.org/a/447770)

The asciicast thumbnail shows part of the bug, but check this out:

image
Only an abort fixes that, and there are 5k+ changes so it will be a looong time before I fix it successfully. Any ideas on this?

**Comment by jsjoeio** [comment]
Looks like @code-asher is here to save the day! He turned on a feature that lets us use GitHub's UI to update your branch. As long as all CI passes, we should be good to merge!

**Comment by im-coder-lg** [comment]
Kinda expected merging through GitHub but thought of using the CLI. I know the GitHub method but didn't understand rebasing fully.

**Comment by im-coder-lg** [comment]
The check failed as usual. Can you revert the commit? Or can _you_ make the PR instead of me? You can make the branch, I'll send the code here, and you put it there, start the PR, and I'll review! I have observed, whenever a maintainer/collaborator makes the PRs, they get approved easily and the checks work as expected. So, I think if this doesn't work as expected till the checks issue is resolved, I'll close the PR and send the code to you, and you can make the PR on this. Is this a better idea?

**Comment by im-coder-lg** [comment]
Ah, fixed! Now if the docs issue is fixed(or maybe not), then the PR is ready to be merged!

**Comment by im-coder-lg** [comment]
So, about the docs, what should I do to fix it? I think if you gain access to the fork of mine, you could fix it. For now, I will try studying the Docs CI and find the issue out.

**Comment by im-coder-lg** [comment]
Aha! The first issue here is: https://github.com/cdr/code-server/blob/94b2774f8c9fb120c19232d2739616e447f9b89d/.github/workflows/docs-preview.yaml#L28-L35

The logs in the checks page is:
```console
Run actions/checkout@v2
  with:
    repository: cdr/m
    ref: refs/heads/master
    submodules: true
    fetch-depth: 0
    ssh-strict: true
    persist-credentials: true
    clean: true
    lfs: false
Error: Input required and not supplied: token
```

The next issue is this: https://github.com/cdr/code-server/blob/94b2774f8c9fb120c19232d2739616e447f9b89d/.github/workflows/docs-preview.yaml#L72-L81

Console logs: 
```console
Run ./ci/scripts/github_deployment.sh update
  ./ci/scripts/github_deployment.sh update
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_DEPLOYMENT: 
    GITHUB_TOKEN: ***
    DEPLOY_STATUS: skipped
    DEPLOY_URL: 
/home/runner/work/_temp/cf73eb14-f128-4001-941c-a1088042e7f8.sh: line 1: ./ci/scripts/github_deployment.sh: No such file or directory
Error: Process completed with exit code 127.
```


The final one is this:
https://github.com/cdr/code-server/blob/94b2774f8c9fb120c19232d2739616e447f9b89d/.github/workflows/docs-preview.yaml#L83-L94
Console logs:
```console
Run marocchino/sticky-pull-request-comment@v2
  with:
    header: codercom-preview-docs
    message: ✨ Coder.com for PR #4419 deployed! It will be updated on every commit.
  
  * _Host_: /docs/code-server
  * _Last deploy status_: skipped
  * _Commit_: f25768a6c21c0ecf646204d3cdf74711f7182576
  * _Workflow status_: https://github.com/cdr/code-server/actions/runs/1435646749
  
    append: false
    recreate: false
    delete: false
    hide_details: false
    hide: false
    hide_and_recreate: false
    hide_classify: OUTDATED
    GITHUB_TOKEN: ***
Error: Resource not accessible by integration
```


How can we fix this other than adding the secret tokens(which I don't want to happen since it's a privacy leak of Coder)?

Maybe if this persists, a maintainer only can finish this successfully.

**Review by jsjoeio — APPROVED** [review]

**Comment by jsjoeio** [comment]
@im-coder-lg nice investigating! Yeah, it's documented [here](https://github.com/cdr/code-server/issues/4027). I need to spend some time soon fixing this - sorry for all your trouble and thanks again for sending this PR ♥️

**Pull request merged**